GameLib/AbstractPlayer.cpp

```diff
@@ -0,0 +1,89 @@
+#include "AbstractPlayer.h"
+#include "AbstractPlayer.h"
+#include <exception>
+
+AbstractPlayer::AbstractPlayer(Coordinate startCoords, const std::vector<ConsoleUtils::Colors> &player_color)
+    : m_coordinate(std::move(startCoords))
+{
+    if (player_color.empty())
+    {
+        throw std::runtime_error("player has to be colored!");
+    }
+    m_playerCharacter = ConsoleUtils::colorString("X", player_color);
+}
+
+AbstractPlayer::~AbstractPlayer() = default; // selbst pure virtual destruktoren müssen definiert werden.
+
+Coordinate AbstractPlayer::getCoordinate()
+{
+    return m_coordinate;
+}
+
+std::string AbstractPlayer::toString()
+{
+    return m_playerCharacter;
+}
+
+bool AbstractPlayer::hasVerticalBarrier(Coordinate &coordinate)
+{
+    for (Barrier barrier : m_barriers)
+    {
+
+        //if (coordinate == barrier.getStartCoord())
+        //{
+        //    if (coordinate.x() == barrier.getEndCoord())
+        //        return true;
+        //}
+    }
+    return false;
+}
+
+bool AbstractPlayer::fieldCheck(const Coordinate &coordinate)
+{
+    if (coordinate.x() >= 0 && coordinate.y() >= 0 && coordinate.x() <= 8 && coordinate.y() <= 8)
```

Comment:
8 ist eine magic number, mach irgendwo ein `costexpr std::size_t gamefield_size` rein und änder alles wo die 8 als spielfeldgröße reinkommt in `gamefield_size`

Reply:
gitbs sogar schon:
```c++
    static constexpr int s_width{9};
    static constexpr int s_height{9};
```